### PR TITLE
Oracle Analytics Server 6.4 public dockerfiles

### PR DESCRIPTION
**Pull request by padambengani**
Made additional fixes as per the comments in the older pull request - https://github.com/oracle/docker-images/pull/2431

### CODE REVIEW

**Comment by Djelibeybi** [comment]
Please resolve the remaining linter issues.

**Comment by Djelibeybi** [comment]
You'll need to make your Oracle email address publicly visible in your GitHub profile settings to stop the OCA Bot from flagging your commits. Please review the "[publicizing your organization membership](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-your-membership-in-organizations/publicizing-or-hiding-organization-membership)" documentation for instructions on how to do so.

**Comment by Djelibeybi** [comment]
If you update the `main` branch of your fork, you'll be able to use the newly added `lint` script to run the linters locally before you push your commits too. See the contribution guide also in the root of the repo for more info.

**Comment by padambengani** [comment]
@Djelibeybi I have fixed the remaining lint issues and also myself & hemala have publicized our oracle membership in our github profiles.

File: OracleAnalytics/dockerfiles/6.4/Dockerfile
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/6.4/Dockerfile:26]: The `MAINTAINER` tag is deprecated. Please replace this with the required `LABEL` directives as documented in the [contribution guide](https://github.com/oracle/docker-images/blob/main/CONTRIBUTING.md).

File: OracleAnalytics/dockerfiles/6.4/Dockerfile
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/6.4/Dockerfile:57]: You're already using `USER oracle` from line 52 so this line is redundant.

File: OracleAnalytics/patches/6.4-patch/Dockerfile
Comment (Djelibeybi) [inline OracleAnalytics/patches/6.4-patch/Dockerfile:39]: The `MAINTAINER` tag is deprecated. Please use `LABEL` directives as documented in the [contribution guide](https://github.com/oracle/docker-images/blob/main/CONTRIBUTING.md).

File: OracleAnalytics/patches/6.4-patch/Dockerfile
Comment (Djelibeybi) [inline OracleAnalytics/patches/6.4-patch/Dockerfile:72]: Please combine these into a single `COPY` directive to reduce layer count.

File: OracleAnalytics/patches/6.4-patch/Dockerfile
Comment (Djelibeybi) [inline OracleAnalytics/patches/6.4-patch/Dockerfile:120]: Please combine these into a single `RUN` directive to reduce layer count and size.

File: OracleAnalytics/patches/6.4-patch/Dockerfile
Comment (Djelibeybi) [inline OracleAnalytics/patches/6.4-patch/Dockerfile:123]: You are already using `USER oracle` from line 76. This line is redundant.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

**Comment by Djelibeybi** [comment]
It would also be great if you could pull the latest commits to the `main` branch of your fork, then rebase this branch on top of the updated `main` branch so that the PR picks up the updated GitHub Actions checks.

**Comment by Djelibeybi** [comment]
I rebased your branch for you. You'll need to run `git pull` to bring any local clones of the repo up-to-date though.

**Comment by padambengani** [comment]
@Djelibeybi I have made the requested changes (haven't pushed them yet), but still when I run the lint script locally, then its failing with following two errors - 

`2022-11-30 15:57:49 [INFO]   File:[/tmp/lint/OracleAnalytics/dockerfiles/6.4/Dockerfile]
2022-11-30 15:57:49 [ERROR]   Found errors in [hadolint] linter!
2022-11-30 15:57:49 [ERROR]   Error code: 1. Command output:
------
/tmp/lint/OracleAnalytics/dockerfiles/6.4/Dockerfile:53 DL3003 warning: Use WORKDIR to switch to a directory


2022-11-30 15:57:49 [INFO]   File:[/tmp/lint/OracleAnalytics/patches/6.4-patch/Dockerfile]
2022-11-30 15:57:49 [ERROR]   Found errors in [hadolint] linter!
2022-11-30 15:57:49 [ERROR]   Error code: 1. Command output:
------
/tmp/lint/OracleAnalytics/patches/6.4-patch/Dockerfile:68 DL3003 warning: Use WORKDIR to switch to a directory`

Please suggest what should be done for these errors as using WORKDIR to switch directory instead of using cd would increase number of layers (esp. for commands with many cd).

**Comment by Djelibeybi** [comment]
> Please suggest what should be done for these errors as using WORKDIR to switch directory instead of using cd would increase number of layers (esp. for commands with many cd).

You don't need to use `cd` at all. Your commands can be modified to specify the right path instead of relying on being run in that path.

For example, in your main `Dockerfile`, use this:

```Dockerfile
RUN unzip "/u01/$BI_DISTRO_ZIP1" && \
    "$JAVA_HOME/bin/java" -jar "/u01/$BI_INSTALLER1" -silent -responseFile /u01/install.file \
      -invPtrLoc /u01/oraInst.loc -ignoreSysPrereqs -force -novalidation ORACLE_HOME="$ORACLE_HOME" \
      INSTALL_TYPE="Oracle Analytics" && \
    rm "/u01/$BI_DISTRO_ZIP1" "/u01/$BI_INSTALLER1" /u01/oraInst.loc /u01/install.file
```

Then, for your patching `Dockerfile` you can change each patch block to something like this:

```Dockerfile
RUN ...
    ...
    unzip "/u01/$PATCH_PKG_33958532" && \
    "$ORACLE_HOME/OPatch/opatch" apply -silent /u01/33958532 && \
    rm "/u01/$PATCH_PKG_33958532" && \
    ...
```

Given that you install `unzip` in your main `Dockerfile`, it makes sense to use it to unpack your patches in your patching `Dockerfile` too.

**Comment by padambengani** [comment]
@Djelibeybi I have now made the suggested changes to rid of "cd" in the RUN commands used in Dockerfiles and tested the same.

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:185]: Never use `chmod 777` as this is insecure. You should be using `700` and ensuring that the user inside the container runs with the same UID/GID as the user that owns the files on the filesystem.

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:135]:
```suggestion
* `ADMIN_USERNAME` - WebLogic admin username for the new domain
* `ADMIN_PASSWORD` - WebLogic admin password
* `DB_HOST` - Host name for database into which to create schemas
* `DB_PORT` - Database listener port
* `DB_SERVICE` - Database instance service name
* `DB_USERNAME` - Database sysdba username
* `DB_PASSWORD` - Database sysdba password
* `SCHEMA_PREFIX` - Schema prefix for new schemas in the new domain. The schema prefix must be unique. If it’s already in use, domain creation fails.
* `SCHEMA_PASSWORD` - Password for all new schemas created

The following variables are predefined:

`ORACLE_HOME` is `/u01/oracle`.
`DOMAIN_HOME` is `/u01/oracle/user_projects/domains/bi`.
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:105]:
```suggestion
See the [Oracle Analytics patched image documentation](./patches/6.4-patch) for details.
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:77]:
```suggestion
Before you can build a BI image, you will need to build the [Oracle Java](https://github.com/oracle/docker-images/tree/master/OracleJava) image and the [Oracle Fusion Middleware](https://github.com/oracle/docker-images/tree/master/OracleFMWInfrastructure) image or pull pre-built images of each from the [Oracle Container Registry](https://container-registry.oracle.com).
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:99]: Please also provide a `docker build` sample command for users who don't want to use this script.

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:154]:
```suggestion
* `9500` - WebLogic AdminServer
* `9502` - WebLogic `bi_server1` managed server, hosting BI JEE apps
* `9508` - OBI Cluster Controller, providing access to OBI Server system component
* `9514` - OBI Server system component
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:218]:
```suggestion
using `docker run -v` or container secrets.
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:232]:
```suggestion
The BI image supports using `docker-compose` to start the BI and DB containers together.
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:250]:
```suggestion
Therefore, if you create a container using an existing data volume for `/u01/oracle/user_projects`,
```

File: OracleAnalytics/README.md
Comment (Djelibeybi) [inline OracleAnalytics/README.md:261]:
```suggestion
3. Container health check isn't implemented.
```

File: OracleAnalytics/dockerfiles/6.4/createAndStartDomain.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/6.4/createAndStartDomain.sh:148]: Either do this, or remove this from the file. Do not leave TODO items in published files.

File: OracleAnalytics/dockerfiles/6.4/createAndStartDomain.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/6.4/createAndStartDomain.sh:204]: Either do these things or remove them from this file. Do not leave TODO items in published files

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:16]:
```suggestion
Builds a container image for Oracle Analytics.
```

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:1]: Please rename this file to `buildContainerImage.sh` and update all references accordingly.

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:21]:
```suggestion
   -c: enables image layer cache during build
```

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:48]: This doesn't match your actual version of `6.4`.

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:80]: Given the default version is wrong, this will fail if omitted.

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:134]:
```suggestion
  Oracle Analytics container image for version $VERSION is ready: 
```

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (Djelibeybi) [inline OracleAnalytics/dockerfiles/buildDockerImage.sh:129]: Please review the updated `buildContainerImage.sh` scripts used by the Database team that support both Docker and Podman container runtimes and update this script to do the same: https://github.com/oracle/docker-images/blob/main/OracleDatabase/SingleInstance/dockerfiles/buildContainerImage.sh#L52-L99

File: OracleAnalytics/patches/6.4-patch/README.md
Comment (Djelibeybi) [inline OracleAnalytics/patches/6.4-patch/README.md:4]: Why are these patches necessary?

File: OracleAnalytics/patches/6.4-patch/README.md
Comment (Djelibeybi) [inline OracleAnalytics/patches/6.4-patch/README.md:12]: You need to make it clear that this option is only available to customers who have the appropriate paid licenses and thus access to My Oracle Support.

**Review by Djelibeybi — CHANGES_REQUESTED** [review]
Please also review the line breaks in all of your Markdown files because your lines are extremely short. You should be able to significantly reduce the number of line breaks and thus the length of each file. If you're concerned about the Markdown linter, the line length used by our checks is different to the default used by Markdownlint and is much longer. The `scripts/lint` tool should be used to check as it uses the right configuration.

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (padambengani) [reply OracleAnalytics/dockerfiles/buildDockerImage.sh:129]: @Djelibeybi Is it ok to add support for Podman container later? i.e. I'll resolve all other issues in this pull request and once this pull request is merged then I'll create a new ticket for myself to add support for Podman and will create a new pull request after testing the same. Please let me know if thats fine.

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (padambengani) [reply OracleAnalytics/dockerfiles/buildDockerImage.sh]: fixed the default version

File: OracleAnalytics/dockerfiles/buildDockerImage.sh
Comment (padambengani) [reply OracleAnalytics/dockerfiles/buildDockerImage.sh:80]: Fixed the default version

File: OracleAnalytics/README.md
Comment (padambengani) [reply OracleAnalytics/README.md]: Fixed

File: OracleAnalytics/README.md
Comment (padambengani) [reply OracleAnalytics/README.md]: Fixed as per the suggestion.

File: OracleAnalytics/patches/6.4-patch/README.md
Comment (padambengani) [reply OracleAnalytics/patches/6.4-patch/README.md:4]: These are the patches which are recommended by OAS Marketplace team to customers and these patches have various critical patches related to security, etc. (across various components like FMW, OPSS, etc.).

File: OracleAnalytics/README.md
Comment (padambengani) [reply OracleAnalytics/README.md:61]: done.

**Review by Djelibeybi — APPROVED** [review]

**Pull request merged**
